From 967aee7aad2accfb38d91ab56b6c5e91d86a2722 Mon Sep 17 00:00:00 2001
From: Stefan 'psYchotic' Zwanenburg <stefanhetzwaantje@gmail.com>
Date: Wed, 12 Oct 2011 01:16:18 +0000
Subject: dock: Fix popup menus

The dock popup menus weren't working because a signal had been renamed
('popup' -> 'open-state-changed').

https://bugzilla.gnome.org/show_bug.cgi?id=661484
---
diff --git a/extensions/dock/extension.js b/extensions/dock/extension.js
index 47cd76f..240f8a8 100644
--- a/extensions/dock/extension.js
+++ b/extensions/dock/extension.js
@@ -682,7 +682,7 @@ DockIcon.prototype = {
             this._menu.connect('activate-window', Lang.bind(this, function (menu, window) {
                 this.activateWindow(window);
             }));
-            this._menu.connect('popup', Lang.bind(this, function (menu, isPoppedUp) {
+            this._menu.connect('open-state-changed', Lang.bind(this, function (menu, isPoppedUp) {
                 if (!isPoppedUp){
                     //Restore value of autohidedock
                     this._dock._restoreHideDock();
@@ -776,7 +776,6 @@ DockIconMenu.prototype = {
         this._source = source;
 
         this.connect('activate', Lang.bind(this, this._onActivate));
-        this.connect('open-state-changed', Lang.bind(this, this._onOpenStateChanged));
 
         this.actor.add_style_class_name('dock-menu');
 
--
cgit v0.9.0.2
